alchemist-incarnation-collektive/src/main/kotlin/it/unibo/alchemist/CollektiveIncarnation.kt

```diff
@@ -58,39 +84,77 @@ class CollektiveIncarnation<P> : Incarnation<Any?, P> where P : Position<P> {
         time: TimeDistribution<Any?>,
         actionable: Actionable<Any?>,
         additionalParameters: String?,
-    ): Condition<Any?> = object : AbstractCondition<Any>(requireNotNull(node)) {
-        override fun getContext() = Context.LOCAL
-        override fun getPropensityContribution(): Double = 1.0
-        override fun isValid(): Boolean = true
-    }
+    ): Condition<Any?> =
+        object : AbstractCondition<Any>(requireNotNull(node)) {
+            override fun getContext() = Context.LOCAL
+
+            override fun getPropensityContribution(): Double = 1.0
+
+            override fun isValid(): Boolean = true
+        }
 
     override fun createReaction(
         randomGenerator: RandomGenerator,
         environment: Environment<Any?, P>,
         node: Node<Any?>,
         timeDistribution: TimeDistribution<Any?>,
         parameter: String,
-    ): Reaction<Any?> = Event(node, timeDistribution).also {
-        it.actions = ListSet.of(
-            createAction(randomGenerator, environment, node, timeDistribution, it, parameter),
-        )
-    }
+    ): Reaction<Any?> =
+        Event(node, timeDistribution).also {
+            it.actions =
+                ListSet.of(
+                    createAction(randomGenerator, environment, node, timeDistribution, it, parameter),
+                )
+        }
 
     override fun createTimeDistribution(
         randomGenerator: RandomGenerator,
         environment: Environment<Any?, P>,
         node: Node<Any?>?,
         parameter: String?,
-    ): TimeDistribution<Any?> = parameter?.toDoubleOrNull().let { frequency ->
-        val actualFrequency = frequency ?: 1.0
-        DiracComb(DoubleTime(randomGenerator.nextDouble(0.0, 1.0 / actualFrequency)), actualFrequency)
-    }
+    ): TimeDistribution<Any?> =
+        parameter?.toDoubleOrNull().let { frequency ->
+            val actualFrequency = frequency ?: 1.0
+            DiracComb(DoubleTime(randomGenerator.nextDouble(0.0, 1.0 / actualFrequency)), actualFrequency)
+        }
 
     override fun createNode(
         randomGenerator: RandomGenerator,
         environment: Environment<Any?, P>,
         parameter: String?,
-    ): Node<Any?> = GenericNode(environment).also {
-        it.addProperty(CollektiveDevice(environment, it, DoubleTime(parameter?.toDoubleOrNull() ?: 0.0)))
+    ): Node<Any?> =
+        GenericNode(environment).also {
+            it.addProperty(
+                CollektiveDevice(
+                    environment,
+                    it,
+                    parameter?.toDoubleOrNull().let { p ->
+                        if (p != null) DoubleTime(p) else null
+                    },
+                ),
+            )
+        }
```

Comment:
I prefer naming the outer lambda parameters instead of the inner ones when multiple nested lambda.